shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/test/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/ClusterContextManagerCoordinatorTest.java

```diff
@@ -250,4 +258,63 @@ private OptimizerContext createOptimizerContext() {
         when(result.getFederationMetaData().getSchemas()).thenReturn(schemas);
         return result;
     }
+    
+    @Test
+    public void assertRenewPrimaryDSNames() {
+        PrimaryStateChangedEvent mockPrimaryStateChangedEvent = mock(PrimaryStateChangedEvent.class);
+        QualifiedSchema qualifiedSchema = mock(QualifiedSchema.class);
+        when(qualifiedSchema.getSchemaName()).thenReturn("test_schema");
+        when(mockPrimaryStateChangedEvent.getQualifiedSchema()).thenReturn(qualifiedSchema);
+        ShardingSphereRuleMetaData mockShardingSphereRuleMetaData = mock(ShardingSphereRuleMetaData.class);
+        Collection<ShardingSphereRule> rules = new LinkedList<>();
+        StatusContainedRule mockStatusContainedRule = mock(StatusContainedRule.class);
+        rules.add(mockStatusContainedRule);
+        when(mockShardingSphereRuleMetaData.getRules()).thenReturn(rules);
+        ShardingSphereMetaData mockShardingSphereMetaData = mock(ShardingSphereMetaData.class);
+        when(mockShardingSphereMetaData.getRuleMetaData()).thenReturn(mockShardingSphereRuleMetaData);
+        contextManager.getMetaDataContexts().getMetaDataMap().put("test_schema", mockShardingSphereMetaData);
+        coordinator.renew(mockPrimaryStateChangedEvent);
+        verify(mockStatusContainedRule, times(1)).updateStatus(any());
+    }   
+    
+    @Test
+    public void assertRenewInstanceStatus() {
+        StateEvent mockStateEvent = mock(StateEvent.class);
```

Comment:
Is it necessary to mock `StateEvent`? If the object is easy to create, please consider about using `new` at first priority.

Reply:
+1. I'll update the code